src/charm.py

```diff
@@ -459,20 +459,21 @@ def _set_alerts(self, container) -> bool:
                 alert rule files need to be created. This container
                 must be in a pebble ready state.
 
-        Returns: A boolean indicating if new alert rules were pushed.
+        Returns: A boolean indicating if new or different alert rules were pushed.
         """
+        alert_rules_changed = True
         metrics_consumer_alerts = self.metrics_consumer.alerts()
         remote_write_alerts = self.remote_write_provider.alerts()
-
         alerts_hash = sha256(str(metrics_consumer_alerts) + str(remote_write_alerts))
+
         if alerts_hash == self._stored.alerts_hash:
-            return False
+            alert_rules_changed = False
 
         container.remove_path(RULES_DIR, recursive=True)
```

Comment:
Should we add a comment here?

```suggestion
        # Pushing files every time for situations such as cluster restart:
        # Relation data and stored state match, but files haven't been written yet.
        container.remove_path(RULES_DIR, recursive=True)
```